socketio/src/client/raw_client.rs

```diff
@@ -306,51 +319,95 @@ impl RawClient {
+            } else {
+                Err(Error::InvalidPacket())
+            }
+        } else {
+            Ok(Payload::Json(value.to_owned()))
+        }
+    }
+
+    fn binary_payloads(vec: &Vec<Value>, attachments: &Option<Vec<Bytes>>) -> Result<Payload> {
+        let mut vec_payload = vec![];
+        for value in vec {
+            if value.get("_placeholder").is_some() {
+                let index = value
+                    .get("num")
+                    .ok_or(Error::InvalidPacket())?
+                    .as_u64()
+                    .ok_or(Error::InvalidPacket())? as usize;
+                if let Some(atts) = attachments {
+                    let b = atts.get(index).ok_or(Error::InvalidPacket())?.to_owned();
+                    vec_payload.push(RawPayload::Binary(b));
+                } else {
+                    return Err(Error::InvalidPacket());
+                }
+            } else {
+                vec_payload.push(RawPayload::Json(value.to_owned()))
+            }
+        }
+        Ok(Payload::Multi(vec_payload))
+    }
+
     /// A method for handling the Event Client Packets.
     // this could only be called with an event
     fn handle_event(&self, packet: &Packet) -> Result<()> {
-        // unwrap the potential data
-        if let Some(data) = &packet.data {
-            // the string must be a valid json array with the event at index 0 and the
-            // payload at index 1. if no event is specified, the message callback is used
-            if let Ok(serde_json::Value::Array(contents)) =
-                serde_json::from_str::<serde_json::Value>(data)
-            {
-                let event: Event = if contents.len() > 1 {
-                    contents
-                        .get(0)
-                        .map(|value| match value {
-                            serde_json::Value::String(ev) => ev,
-                            _ => "message",
-                        })
-                        .unwrap_or("message")
-                        .into()
-                } else {
-                    Event::Message
-                };
-                self.callback(
-                    &event,
+        // the string must be a valid json array with the event at index 0 and the
+        // payload at index 1. if no event is specified, the message callback is used
+        if let Some(serde_json::Value::Array(contents)) = &packet.data {
+            let event: Event = if contents.len() > 1 {
+                contents
+                    .get(0)
+                    .map(|value| match value {
+                        serde_json::Value::String(ev) => ev,
+                        _ => "message",
+                    })
+                    .unwrap_or("message")
+                    .into()
+            } else {
+                Event::Message
+            };
+
+            let payload = if contents.len() > 1 {
+                Payload::Multi(contents[1..].iter().map(|v| v.to_owned().into()).collect())
```

Comment:
nit: alternative. No difference at all, just if you feel like using iterator's functions
```suggestion
                Payload::Multi(contents.iter().skip(1).cloned().map(RawPayload::from).collect())
```